waku/waku_metadata/protocol.nim

```diff
@@ -27,57 +27,51 @@ const RpcResponseMaxBytes* = 1024
 type
   WakuMetadata* = ref object of LPProtocol
     clusterId*: uint32
-    shards*: seq[uint32]
+    shards*: HashSet[uint32]
 
 proc respond(m: WakuMetadata, conn: Connection): Future[Result[void, string]] {.async, gcsafe.} =
-  try:
-    await conn.writeLP(WakuMetadataResponse(
-      clusterId: some(m.clusterId),
-      shards: m.shards
-      ).encode().buffer)
-  except CatchableError as exc:
-    return err(exc.msg)
+  let response = WakuMetadataResponse(
+    clusterId: some(m.clusterId),
+    shards: toSeq(m.shards)
+  )
+  
+  let res = catch: await conn.writeLP(response.encode().buffer)
+  if res.isErr():
+    return err(res.error.msg)
 
   return ok()
 
 proc request*(m: WakuMetadata, conn: Connection): Future[Result[WakuMetadataResponse, string]] {.async, gcsafe.} =
-  var buffer: seq[byte]
-  var error: string
-  try:
-    await conn.writeLP(WakuMetadataRequest(
-      clusterId: some(m.clusterId),
-      shards: m.shards,
-    ).encode().buffer)
-    buffer = await conn.readLp(RpcResponseMaxBytes)
-  except CatchableError as exc:
-    error = $exc.msg
-  finally:
-    # close, no more data is expected
-    await conn.closeWithEof()
+  let request = WakuMetadataRequest(clusterId: some(m.clusterId), shards: toSeq(m.shards))
+
+  let writeRes = catch: await conn.writeLP(request.encode().buffer)
```

Comment:
same, i think it can raise an exception

Reply:
Same here `catch:` take care of the exception